lib/teleterm/clusters/cluster_file_transfer.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2022 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clusters
+
+import (
+	"context"
+	"io"
+	"os"
+	"sync"
+	"time"
+
+	"github.com/gravitational/teleport/lib/sshutils/sftp"
+	api "github.com/gravitational/teleport/lib/teleterm/api/protogen/golang/v1"
+
+	"github.com/gravitational/trace"
+)
+
+type FileTransferProgressSender = func(progress *api.FileTransferProgress) error
+
+func (c *Cluster) TransferFile(ctx context.Context, request *api.FileTransferRequest, sendProgress FileTransferProgressSender) error {
+	var config *sftp.Config
+	var configErr error
+
+	direction := request.GetDirection()
+	if direction == api.FileTransferDirection_FILE_TRANSFER_DIRECTION_UNSPECIFIED {
```

Comment:
It's worth being more careful with gRPC enums. If for some reason, a newer client was connected to this service, and that client sent a new direction not yet implemented in the server, this would result in a nil `config`. Whilst this is unlikely here since the server and the client are pretty tightly bound here, I think it's a good habit to have. Switching to a case statement with a default is probably safer.

Reply:
Good idea, switched to switch/case statement and moved to a new function.